src/rules/linalg/dense.jl

```diff
@@ -69,3 +69,34 @@ end
 frule(::typeof(tr), x) = (tr(x), Rule(Δx -> tr(extern(Δx))))
 
 rrule(::typeof(tr), x) = (tr(x), Rule(ΔΩ -> Diagonal(fill(ΔΩ, size(x, 1)))))
+
+#####
+##### Binary operations
+#####
+
+const BINARY_LINALG_OPS = [
+    (*, AbstractArray, AbstractArray,
+        :(Ȳ * B'),
+        :(A' * Ȳ)),
+    (/, AbstractArray, AbstractArray,
+        :(Ȳ / transpose(B)),
+        :(-transpose(Y) * (Ȳ / transpose(B)))),
+    (\, AbstractArray, AbstractArray,
+        :(-(transpose(A) \ Ȳ) * transpose(Y)),
+        :(transpose(A) \ Ȳ)),
+    #(norm, AbstractArray, Number,
```

Comment:
Before merging I had wanted to figure out why `norm` wasn't working and add tests for rectangular matrices.